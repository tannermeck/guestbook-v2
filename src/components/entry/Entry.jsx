import './entry.css';

const Entry = ({ entries }) => {
    return (
        <section className='entryContainer'>
            {entries.map(entry => (
                    <div className='entry' key={`${entry.name}-${entry.entry}`}>
                        <p className='entryName'>-{entry.name}</p>
                        <p className='entryText'>{entry.entry}</p>
                    </div>
                ))
            }
        </section>
    )
}
export default Entry;