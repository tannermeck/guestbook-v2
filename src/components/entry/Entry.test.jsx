import { screen, render } from '@testing-library/react';
import Entry from './Entry';

it('should render entries', async () => {
    const entries = [{guest: 'kittle', entry: 'Thank you'}]
    const { container } = render(<Entry entries={entries}/>)
    
    const guest = await screen.findByText('kittle', {exact: false})
    const entry = await screen.findByText('Thank you')
    
    expect(guest).toBeInTheDocument();
    expect(entry).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});