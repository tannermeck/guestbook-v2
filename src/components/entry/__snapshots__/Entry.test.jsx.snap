// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render entries 1`] = `
<div>
  <section
    class="entryContainer"
  >
    <div
      class="entry"
    >
      <p
        class="entryName"
      >
        -
        kittle
      </p>
      <p
        class="entryText"
      >
        Thank you
      </p>
    </div>
  </section>
</div>
`;
